src/main/File/Services/DownloadFileService.java

```diff
@@ -0,0 +1,152 @@
+package File.Services;
+
+import Config.Message;
+import Config.Service;
+import Database.File.FileDao;
+import File.File;
+import File.FileMessage;
+import File.FileType;
+import Security.EncryptionController;
+import User.UserType;
+import com.mongodb.client.MongoDatabase;
+import lombok.extern.slf4j.Slf4j;
+import org.bson.types.ObjectId;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.Objects;
+import java.util.Optional;
+
+@Slf4j
+public class DownloadFileService implements Service {
+  MongoDatabase db;
```

Comment:
as stated before, lets remove the MongoDatabase db field and just rely on the DAO